src/macros.js

```diff
@@ -157,6 +175,57 @@ defineMacro("\\global", (context) => {
     }
 });
 
+// \newcommand{\macro}[args]{definition}
+// \renewcommand{\macro}[args]{definition}
+// TODO: Optional arguments: \newcommand{\macro}[args][default]{definition}
+const newcommand = (context, existsOK: boolean, nonexistsOK: boolean) => {
+    let arg = context.consumeArgs(1)[0];
+    if (arg.length !== 1) {
+        throw new ParseError(
+            "\\newcommand's first argument must be a macro name");
+    }
+    const name = arg[0].text;
+
+    // A command is "defined" in KaTeX if it's a macro, function, or symbol.
+    const exists = context.macros.get(name) || functions.hasOwnProperty(name) ||
+        symbols.math.hasOwnProperty(name) || symbols.text.hasOwnProperty(name)
+        || implicitCommands.hasOwnProperty(name);
```

Comment:
I believe this will be useful in many ways, even as a public API! Could you move this to another class or a separate module?

Reply:
Good idea!  It's difficult to make it a public API in `katex` because it depends on the current state of the macros (i.e. the current group).  So I've made it a method of the `MacroExpander` for now, so at least it's available to all macro definitions (defined by JavaScript functions).